src/browser-lib/extra-providers.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable, inject, NgZone, StaticProvider, Inject } from '@angular/core';
+import {
+  ɵBrowserPlatformLocation,
+  PlatformLocation,
+  LocationChangeEvent,
+  DOCUMENT,
+} from '@angular/common';
+
+@Injectable()
+class SingleSpaPlatformLocation extends ɵBrowserPlatformLocation {
+  onPopState(fn: (event: LocationChangeEvent) => void): void {
+    const ngZone = inject(NgZone);
+
+    super.onPopState(event => {
+      // Wrap any event listener into zone that is specific to some application.
+      // The main issue is `back/forward` buttons of browsers, because they invoke
+      // `history.back|forward` which dispatch `popstate` event. Since `single-spa`
+      // overrides `history.replaceState` Angular's zone cannot intercept this event.
+      // Only the root zone is able to intercept all events.
+      ngZone.run(() => fn(event));
+    });
+  }
+}
+
+/**
+ * The `PlatformLocation` class is an "injectee" of the `PathLocationStrategy`,
+ * which creates `Subject` internally for listening on `popstate` events. We want
+ * to provide this class in the most top injector that's used during bootstrapping.
+ */
+export function getSingleSpaExtraProviders(): StaticProvider[] {
+  return [
+    {
+      provide: PlatformLocation,
+      useClass: SingleSpaPlatformLocation,
+      deps: [[new Inject(DOCUMENT)]],
+    },
+  ];
+}
+
+// Providers have to be exported but we don't want our users to consume
+// this class because it's meant to be private.
+export { SingleSpaPlatformLocation as ɵSingleSpaPlatformLocation };
```

Comment:
Why the `ɵ` character? Is that a convention within the Angular community? What is the downside of just exporting it as `SingleSpaPlatformLocation`

Reply:
Yeah, the "theta" symbol is a convention. Same like in that class that we extend `ɵBrowserPlatformLocation`.

> What is the downside

No downside actually, it just means that class is "private" and shouldn't be consumed by our users.

Reply:
Removed theta symbol since the class was used in another file.

Reply:
While testing I encountered another problem.... :monkey_face: 

The `ɵBrowserPlatformLocation` was moved into another package in Angular 9, so:
```ts
import { ɵBrowserPlatformLocation } from '@angular/platform-browser` // Angular 6-8
import { ɵBrowserPlatformLocation } from '@angular/common' // Angular 9
```

They always have some breaking changes in major versions. Yeah, it sux...

What would you propose? For sure we could publish different versions `single-spa-angular@4` where peer dependency would be `@angular/core < 9` and `single-spa-angular@5` with `@angular/core > 9`.

Just thinking aloud....

Reply:
Hmmm, yeah I'd like this fix to be available for as many versions of angular as possible. I think requiring the user to pass in `ɵBrowserPlatformLocation` is the easiest way to accomplish that.